pkg/testutil/prometheus.go

```diff
@@ -146,7 +150,7 @@ func CreateBlock(
 	if err != nil {
 		return id, errors.Wrap(err, "create head block")
 	}
-	defer h.Close()
+	defer runutil.BestEffortErr(nil, &err, h, "TSDB Head")
```

Comment:
Rename this function to `CaptureErrIfNotSet` and change comment to `CaptureErrIfNotSet runs closer...` Also change `wrap` argument name to `format` so that it's consistent with `Sprintf`. Also logger :P.

Reply:
good points